### PR TITLE
Add a feature flag to use a bundle compressed using Zstandard when setting up the default tools

### PR DESCRIPTION
**Pull request by henrymercer**
This PR adds a feature flag to download a CodeQL Bundle compressed using Zstandard when setting up the default tools.

If we determine that `tar` is new enough, we will download the zstd-compressed bundle.

We add a fallback mechanism: if setting up the zstd-compressed bundle fails, we will fall back to the gzip-compressed bundle.

We add PR checks to validate both setting up a zstd-compressed bundle when the feature flag is enabled, and the fallback mechanism.

### Merge / deployment checklist

- [ ] Confirm this change is backwards compatible with existing workflows.
- [ ] Confirm the [readme](https://github.com/github/codeql-action/blob/main/README.md) has been updated if necessary.
- [ ] Confirm the [changelog](https://github.com/github/codeql-action/blob/main/CHANGELOG.md) has been updated if necessary.

### CODE REVIEW

File: pr-checks/checks/zstd-bundle-fallback.yml [hunk 1]
```diff
@@ -0,0 +1,74 @@
+name: "Zstandard bundle fallback"
+description: "Tests the fallback for when downloading a Zstandard-compressed CodeQL Bundle fails"
```
Comment (aeisenberg) [inline pr-checks/checks/zstd-bundle-fallback.yml:2]: Copy-edit

```suggestion
description: "Tests the fallback when downloading a Zstandard-compressed CodeQL Bundle fails"
```

File: pr-checks/checks/zstd-bundle.yml [hunk 2]
```diff
@@ -0,0 +1,63 @@
+name: "Zstandard bundle"
+description: "Tests the feature flag that downloads a Zstandard-compressed CodeQL Bundle by default"
+versions:
+  - linked
+operatingSystems:
+  - ubuntu
```
Comment (aeisenberg) [inline pr-checks/checks/zstd-bundle.yml:6]: Is there a windows and mac test where the zstd bundle is downloaded?

File: src/setup-codeql.ts [hunk 3]
```diff
@@ -722,3 +815,15 @@ function sanitizeUrlForStatusReport(url: string): string {
...
 }
+
+async function useZstdBundle(
+  cliVersion: string,
+  features: FeatureEnablement,
+  tarSupportsZstd: boolean,
+): Promise<boolean> {
+  return (
+    tarSupportsZstd &&
+    semver.gte(cliVersion, "2.19.0") &&
```
Comment (aeisenberg) [inline src/setup-codeql.ts:826]: Can you extract this as a constant and place it somewhere near the declaration of the feature flag in `feature-flags.ts`? This way at least the version and the flag would be within reading distance of each other.

File: pr-checks/checks/zstd-bundle.yml (hunk 2 above)
Comment (henrymercer) [reply pr-checks/checks/zstd-bundle.yml:8]: These are tested in the nightly checks, but we can test here too defensively.

**Review by aeisenberg — APPROVED** [review]
LGTM!

**Review by NlightNFotis — APPROVED** [review]
LGTM

**Pull request merged**